acme/acme/crypto_util.py

```diff
@@ -39,9 +39,11 @@ class SSLSocket(object):  # pylint: disable=too-few-public-methods
     :ivar method: See `OpenSSL.SSL.Context` for allowed values.
 
     """
-    def __init__(self, sock, certs, method=_DEFAULT_TLSSNI01_SSL_METHOD):
+    def __init__(self, sock, cert_selection, alpn_selection=None,
```

Comment:
I appreciate you setting up the plumbing here so we can only serve the cert when it's requested using ALPN when openssl and pyopenssl support it and while this change is probably OK, we've been avoiding making breaking changes in `acme` since we entered beta in 2015. With this in mind, I'm thinking we should revert the `cert -> cert_selection` changes here for now and we can revisit this if desired when this feature is added to our dependencies.

Does this seem reasonable to you?

Reply:
You're perfectly right about breaking changes. I think it's reasonable though to provide this callback feature, so I just refactored it to not break anything. I also changed the order so that the added parameters are the last ones (in case sb specified them as positional).

Is that ok for you?

Reply:
That works for me! It certainly makes it easier for us to do cert selection based on ALPN when it's an option for us.

Thanks for catching the parameter order and moving the new ones to the end.

Only suggested change with this approach is maybe we should make `certs` an optional parameter now with a default value of `None` as it makes the interface a little nicer when you want to use `cert_selection`. If we did this, we should also check that one of `certs` or `cert_selection` is set. You're welcome to ignore this suggestion if you prefer.

Reply:
done